### ISSUE TITLE
Question regarding Probability output

### ISSUE THREAD

**Issue by eddienewton** [callmeray/PointMVSNet#7]
Hi,

I was testing with my own images, which has a different overall scaling.  My depth range is about 0.2 to 2.0.  

My depth output is good, and I'm able to get good results.  However, I'm having issue with the probability output.  My probabilities are noisy, and high probability values do not correspond to good depth values.  

With your flow calc, you're getting 5 probability images, correct?  Is the summation correct?  In the eval code, the red_interval is multiplied by [-2,-1,0,1,2].  So is the middle flow probability ignored?

Is there any scaling I need to look at since my depth range is about 1000x smaller than yours?  

Thanks!

**Comment by eddienewton** [comment]
I was able to fix the probability maps.  In the 'test.py' script, I had changed `model.train()` to `model.eval()`.  I still haven't looked into why the probability maps change, but at least we know the root cause.

**Comment by callmeray** [comment]
Hi,

Really glad to hear your good results on your own images. For my experience, `model.train()` can get better results than `model.eval()`.  The `flow_prob` is computed as the sum of the two probabilities in the neighborhood of the predicted depth. Scaling may need to be changed according to your depth range for better prediction.

**Comment by eddienewton** [comment]
Hi @callmeray,

Yes, I just went ahead and ran more of my images.  The probability is similar to what one gets on the DTU dataset.  It looks like setting eval() affects the batch norms, thus affecting the results.

I had another question regarding probability.  How close should the initial probability (MVSNet) and your probability match?  It appears the PointMVSNet probability isn't as continuous (see attached pic).  The pic is DTU scan 1, image 0.

![prob](https://user-images.githubusercontent.com/43154795/65750447-44d8a400-e0bd-11e9-9952-f0cbfe471d70.jpg)

Finally, can you explain the flow_prob calculation a bit more (if that's ok)?

Thanks!

**Comment by callmeray** [comment]
The initial probability and flow probability do not have to match, since init prob is the prob through whole depth range while flow prob is the prob within the neighborhood.

For example, the prob of the five point hypotheses is [0.1, 0.2, 0.3, 0.3, 0.1], the expected flow is 0.1*-2 + 0.2*-1 + 0.3*1 + 0.1 *2 = 0.1. It's in range [0, 1]. Thus its probability is computed as the summation of the prob at 0 and the prob at 1: 0.3 + 0.3 = 0.6.

**Comment by eddienewton** [comment]
@callmeray thanks for the explanation.  This makes sense in terms of a good approximation for probability.

I'll close this issue as we figured out what's causing the probability disparity with eval().

Thanks again!
